src/backup_observer.py

```diff
@@ -94,3 +96,56 @@ def _on_create_backup_action(self, event: ActionEvent) -> None:
             return
 
         event.set_results({"result": "correct", "backup-id": backup_id})
+
+    def _generate_backup_list_formatted(self, backup_list: typing.List[backup.S3Backup]) -> str:
+        """Generate a formatted string for the backups.
+
+        Args:
+            backup_list: List of backups to create a formatted string for.
+
+        Returns:
+            The formatted string of the backups.
+        """
+        output = [f"{'backup-id':<29s} | {'last-modified':<28s} | {'size':>15s}"]
+        output.append("-" * len(output[0]))
+        for cur_backup in backup_list:
+            output.append(
+                f"{cur_backup.backup_id:<29s}"
+                f" | {cur_backup.last_modified!s:<28s}"
+                f" | {cur_backup.size:>15d}"
+            )
+        return "\n".join(output)
+
+    def _on_list_backups_action(self, event: ActionEvent) -> None:
+        """List backups in S3 configured storage.
+
+        Args:
+            event: Event triggering the list backups action.
+        """
+        try:
+            s3_parameters = backup.S3Parameters(**self._s3_client.get_s3_connection_info())
+        except ValueError:
+            logger.exception("Wrong S3 configuration in list backups action")
```

Comment:
It would be nice to log the exception message for debugging purposes

Reply:
By default, the exception (including the trace) is added with the logger.exception, see https://docs.python.org/3/library/logging.html#logging.Logger.exception